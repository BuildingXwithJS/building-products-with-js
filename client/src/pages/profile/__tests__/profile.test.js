// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import ProfilePage, {Profile} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '0',
  login: 'test',
  registrationDate: new Date(2016, 1, 1, 1, 1, 1, 1),
};

test('# Profile page wrapper', () => {
  const store = mockStore({auth: {user}, users: {user}});
  const wrapper = shallow(<ProfilePage store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Profile page', () => {
  const getUser = (u) => {
    expect(u).toEqual(user);
  };

  const component = (
    <Profile
      getUser={getUser}
      params={user}
      user={user}
      loadedUser={user}
    />
  );

  // test rendering
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});
