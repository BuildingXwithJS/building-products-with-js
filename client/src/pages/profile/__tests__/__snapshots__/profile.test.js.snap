exports[`test # Profile page 1`] = `
<div>
  <Component
    current="/profile/0"
    user={
      Object {
        "id": "0",
        "login": "test",
        "registrationDate": 2016-02-01T00:01:01.001Z,
      }
    } />
  <Connect(User)
    edit={true}
    user={
      Object {
        "id": "0",
        "login": "test",
        "registrationDate": 2016-02-01T00:01:01.001Z,
      }
    } />
</div>
`;

exports[`test # Profile page wrapper 1`] = `
<Profile
  getUser={[Function]}
  loadedUser={
    Object {
      "id": "0",
      "login": "test",
      "registrationDate": 2016-02-01T00:01:01.001Z,
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  user={
    Object {
      "id": "0",
      "login": "test",
      "registrationDate": 2016-02-01T00:01:01.001Z,
    }
  } />
`;
